GitUI/CommandsDialogs/FormEditor.cs

```diff
@@ -116,40 +116,60 @@ private void FormEditor_FormClosing(object sender, FormClosingEventArgs e)
 
         private void toolStripSaveButton_Click(object sender, EventArgs e)
         {
-            try
-            {
-                SaveChanges();
-            }
-            catch (Exception ex)
-            {
-                MessageBox.Show(this, _cannotSaveFile.Text + Environment.NewLine + ex.Message, Strings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
-            }
+            SaveChanges();
         }
 
-        private void SaveChanges()
+        /// <summary>
+        /// Saves Changes
+        /// </summary>
+        /// <param name="rethrowException">Allows Exception to be rethrown for custom handling if saving chang fails</param>
+        private void SaveChanges(bool rethrowException = false)
         {
-            if (!string.IsNullOrEmpty(_fileName))
+            try
             {
-                if (fileViewer.FilePreamble is null || Module.FilesEncoding.GetPreamble().SequenceEqual(fileViewer.FilePreamble))
-                {
-                    File.WriteAllText(_fileName, fileViewer.GetText(), Module.FilesEncoding);
-                }
-                else
+                if (!string.IsNullOrEmpty(_fileName))
```

Comment:
Please invert to reduce nesting

Reply:
This showed up as diff because of the indentation change, which will be reverted anyway (as we both requested).
So I'd say this refactoring is optional.